src/components/Contacts/ContactDetails/ContactDetailsTab/ContactDetailsTab.tsx

```diff
@@ -65,90 +118,190 @@ export const ContactDetailsTab: React.FC<ContactDetailTabProps> = ({
   });
 
   const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { push, query } = useRouter();
+  const [deleteContact, { loading: deleting }] = useDeleteContactMutation();
+  const [dialogOpen, changeDialogOpen] = useState(false);
+
+  const { searchTerm } = query;
+
+  const handleDeleteContact = () => {
+    try {
+      deleteContact({
+        variables: {
+          accountListId,
+          contactId,
+        },
+        update: (cache, { data: deletedContactData }) => {
+          const deletedContactId = deletedContactData?.deleteContact?.id;
+          const query = {
+            query: ContactsDocument,
+            variables: {
+              accountListId,
+              searchTerm,
+            },
+          };
+
+          const dataFromCache = cache.readQuery<ContactsQuery>(query);
+
+          if (dataFromCache) {
+            const data = {
+              contacts: {
+                ...dataFromCache.contacts,
+                nodes: [
+                  ...dataFromCache.contacts.nodes.filter(
+                    (contact) => contact.id !== deletedContactId,
+                  ),
+                ],
+              },
+            };
+            cache.writeQuery({ ...query, data });
+
+            push({
+              pathname: '/accountLists/[accountListId]/contacts',
+              query: { accountListId, searchTerm },
+            });
+            enqueueSnackbar(t('Contact successfully deleted'), {
+              variant: 'success',
+            });
+          }
+        },
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
+  };
+
+  const renderDeleteContactDialog = () => {
+    return (
+      <Dialog
```

Comment:
Are there any simpler modals? Should we make a component for a confirmation dialog so we don't have to have all this HTML and CSS in every component? Seems like it could take a title, body, action label, action callback, and maybe a cancel label if that needs to be customized too.

Reply:
Yeah, I'm going to make a generic modal component and switch this one out for that. Then I'll make a separate pr/ticket to update all the other modals to use the new generic one.